### PR TITLE
Add pipeline publishing section in docs

### LINKED ISSUE #267: Add new documentation section on Pipeline packaging and publication to PyPI as a Wheel
As a new section in the https://scancodeio.readthedocs.io/en/latest/custom-pipelines.html / https://github.com/nexB/scancode.io/blob/main/docs/custom-pipelines.rst chapter (could also be a separate chapter if needed).

The goal is to document how to package and publish one or more custom pipelines (along their pipes) as a library.

### PR DESCRIPTION
**Pull request by AvishrantsSh**
Add a new document section describing the publication of a pipeline package to PyPI.
Closes #267 

Signed-off-by: Avishrant Sharma <avishrants@gmail.com>

### CODE REVIEW

File: docs/custom-pipelines.rst [hunk 1]
```diff
@@ -249,5 +251,54 @@ the entry point to the pipeline under the ``[options.entry_points]`` section.
         pipeline_name = scancodeio_scan_and_report_pipeline.pipelines.scan_and_report:ScanAndReport
 
 .. tip::
-    Take a look at `Google LicenseClassifier pipeline for Scancode.io <https://github.com/nexB/scancode.io-pipeline-glc_scan>`_
+    Take a look at `Google License Classifier pipeline for Scancode.io <https://github.com/nexB/scancode.io-pipeline-glc_scan>`_
     for a complete example on packaging a custom tool as a pipeline.
+
+Publishing Pipeline to PyPI
```
Comment (hyounes4560) [inline docs/custom-pipelines.rst:257]: Pipeline Publishing to PyPI

File: docs/custom-pipelines.rst [hunk 2]
```diff
@@ -249,5 +251,54 @@ the entry point to the pipeline under the ``[options.entry_points]`` section.
...
+---------------------------
+After successfully packaging a pipeline, one may want to distribute it as a plugin via PyPI.
```
Comment (hyounes4560) [inline docs/custom-pipelines.rst:259]: After successfully packaging a pipeline, you may consider distributing it—as a plugin—via PyPI.
Ensure a directory structure similar to the :ref:`pipeline_packaging_example` with all
package files correctly configured.

File: docs/custom-pipelines.rst [hunk 3]
```diff
@@ -249,5 +251,54 @@ the entry point to the pipeline under the ``[options.entry_points]`` section.
...
+
+Next step involves generating the distribution archives for the package. Make sure you have the
+latest version of
+``build`` installed on your system
+
+.. code-block:: bash
+
+    pip install --upgrade build
+
+Now run the following command from the same directory where :guilabel:`pyproject.toml` is located:
```
Comment (hyounes4560) [inline docs/custom-pipelines.rst:275]: Now run the following command from within the same directory where the :guilabel:`pyproject.toml` is located:

File: docs/custom-pipelines.rst [hunk 4]
```diff
@@ -249,5 +251,54 @@ the entry point to the pipeline under the ``[options.entry_points]`` section.
...
+
+.. code-block:: bash
+
+    python -m build
+
+Once completed, you should have two files present in the ``dist`` directory with extensions ``.tar.gz`` and ``.whl``.
```
Comment (hyounes4560) [inline docs/custom-pipelines.rst:281]: Once completed, you should have two files inside the ``dist`` directory with the ``.tar.gz`` and ``.whl`` extensions.

.. note:: 
    Remember to create an account on `PyPI <https://pypi.org/>`_ before uploading your distribution archive to PyPI.

File: docs/custom-pipelines.rst [hunk 5]
```diff
@@ -249,5 +251,54 @@ the entry point to the pipeline under the ``[options.entry_points]`` section.
...
+To upload your distribution archive to PyPI, first create an account on `PyPI <https://pypi.org/>`_.
+
+You can use ``twine`` to upload the package to PyPI. To install twine, use
```
Comment (hyounes4560) [inline docs/custom-pipelines.rst:284]: You can use ``twine`` to upload the package to PyPI. To install twine, run the following command:

File: docs/custom-pipelines.rst [hunk 6]
```diff
@@ -249,5 +251,54 @@ the entry point to the pipeline under the ``[options.entry_points]`` section.
...
+
+.. code-block:: bash
+
+    pip install twine
+
+Upload your package to PyPI as follows
```
Comment (hyounes4560) [inline docs/custom-pipelines.rst:290]: Finally, you can upload your package to PyPI with the next command

File: docs/custom-pipelines.rst [hunk 7]
```diff
@@ -249,5 +251,54 @@ the entry point to the pipeline under the ``[options.entry_points]`` section.
...
+
+.. code-block:: bash
+
+    twine upload dist/*
+
+Once uploaded your pipeline package should be viewable on PyPI, under the name specified by your
```
Comment (hyounes4560) [inline docs/custom-pipelines.rst:296]: Once successfully uploaded, your pipeline package should be viewable on PyPI under the name specified in your
manifest.

File: docs/custom-pipelines.rst [hunk 8]
```diff
@@ -249,5 +251,54 @@ the entry point to the pipeline under the ``[options.entry_points]`` section.
...
+manifest.
+
+To make your pipeline available in ScanCode.io, you need to install the package from PyPI. For
+example, to download the package described in :ref:`pipeline_packaging_example`, use
```
Comment (hyounes4560) [inline docs/custom-pipelines.rst:300]: To make your pipeline available in ScanCode.io, you need to install the package from PyPI. For
For example, to download the package described in the :ref:`pipeline_packaging_example`, run

**Review by hyounes4560 — CHANGES_REQUESTED** [review]

**Comment by hyounes4560** [comment]
FYI @tdruez

File: docs/custom-pipelines.rst (hunk 8 above)
Comment (AvishrantsSh) [reply docs/custom-pipelines.rst]: There was a small typo in the changes you requested (**For** was repeated twice).

**Pull request merged**